### ISSUE TITLE
Question: Using Swarm JS in the browser

### ISSUE THREAD

**Issue by iSasFTW** [MaiaVictor/swarm-js#3]
Hi! I'm wondering what the best way to use swarm-js in the browser is. Do I need to have all of the dependencies (shown on npm) installed, or will it work fine with only src/swarm.js?

On your demo, I see a minified swarm.js, loaded in using `let swarm = Swarm.at("http://swarm-gateways.net");`. Will this work fine anywhere, or is it specific to your demo? (I'm asking, because I might not be able to see all of the files on dev tools)

Thanks!

**Comment by MaiaVictor** [comment]
Oh. Sorry for the delay. I missed this issue. To run on the browser, I think swarm-gateways is the only way currently, since the client won't have Swarm running. You could also host your own swarm gateway, that'd be better. It would be awesome if one day we had a Swarm light-client on JS. Here is how you could use it on the browser:

    var swarm = Swarm.at("url to swarm");
    
    // uploading
    document.getElementById("upload_to_swarm").onclick = function () {
        // could also be pick: "diretory"
        swarm.upload({pick: "file"}).then(function (hash) {
            console.log("Uploaded to swarm! Hash: " + hash);
        });
    }

    // downloading
    swarm.download("swarm_hash").then(function (result) {
        console.log("Downloaded from Swarm:", result);
    });

**Comment by iSasFTW** [comment]
Thanks! I actually figured it out from your demo, but maybe it's worth adding these instructions for in-browser use to the readme.md, so it's easier to find. I used this library in Status' hackathon, and it worked very well for what I was doing. I had a bit of trouble, since I didn't realize the downloads/uploads were async, and just general issues with JS' scope stuff (passing data to the parent function of the callback), but in the end it worked out well. I agree that a light swarm client would be great, and helpful for decentralization. I do believe that Status is planning to add a Swarm client into their app, so dapps and chatbots can use it, and that might be good too.
